_posts/2022-06-01-snitching-on-your-colleagues-using-cloud-magic.md

```diff
@@ -0,0 +1,332 @@
+### Before We Dive In
+Before we dive in there are some nice to know things about the inner workings and structure of a pipeline defined in a yaml file.
+I'll briefly go over some key definitions, so you can follow along nicely when each individual task gets explained.
+- Triggers :
+  - Triggers are (like the word implies) what sets off your pipeline
+  - These can be changes in a branch like ***master*** or ***develop*** but could also be specific events happening in another pipeline such as failed jobs/stages or a specific variable value
+- Variables :
+  - A hardcoded variable defined at the start of your pipeline
+  - A variable group containing secret values like tokens that can not be acquired using service connections
+  - Both of these options can be defined at the same time or individually and used anywhere in the pipeline including in additional task commands if the task allows it.
+- Stages : 
+  - Defines what steps your pipeline should take in what order.
+  - Encapsulation of ***stage*** sections
+- Stage :
+  - A "section" of your pipeline
+  - Can be given a name of your choosing eg: Test,Docker,CloudSetup ...
+  - Encapsulation for ***jobs***
+- Jobs : 
+  - Encapsulation of ***job*** sections
+- Job :
+  - Can be given a name of your choosing just like the ***stage*** section
+  - Can be given a ***pool*** attribute to define which agent OS has to run this job
+  - Encapsulation of one or more ***task*** sections
+- Task :
+  - A pre-built or custom-made task to be performed on your pipeline run.
+  - has a variety of attributes that can be manually filled up such as credentials or dictating your preferred working directories
+- Agents :
+  - A machine hosted by the cloud provider (Azure in our case) that runs on a specific OS
+  - Defined in the ***pool*** attribute of a ***job*** or at the start of a pipeline.
+  - Mostly a clean slate with only some essential software pre-installed eg: Docker
+  - Gets wiped after use in order to accommodate the next job.
+
+All of the above gets combined into a structure that resembles the image below.
+
+<img class="p-image" src="{{ '/img/2022-06-01-snitching-on-your-colleagues-using-cloud-magic/ExampleStructure.png' | prepend: site.baseurl }}" class="image fit" style="margin:0px auto; max-width: 50%;">
+
+### Basic Tasks
+Following are the pipeline tasks used to test our java project, generate a test rapport and then build it using Maven.
+#### Maven Test
+To test our application we will be using JUnit because of the pre-existing support given by Azure. This will also generate a test rapport during each pipeline run based on the unit tests defined in our project.
```

Comment:
```suggestion
To test our application we will be using JUnit because of the pre-existing support given by Azure. 
This will also generate a test rapport during each pipeline run based on the unit tests defined in our project.
```